### ISSUE TITLE
MD014 does not warn for fenced code blocks (just indented ones)

### ISSUE THREAD

**Issue by DavidAnson** [markdownlint/markdownlint#106]
See DavidAnson/markdownlint@5e3d9f8df420a2b340fbfa97146eec916f9676eb for a test file that I expect to trigger MD014 in the Ruby implementation but that does not.

In this case, one could argue whether MD014 should be applied to the first line of the fenced code block or the first line of its content. I chose the first line of the block because that seemed to match the spirit of the rule (i.e., it isn't meant to trigger for each '$' violation), but I could be convinced to change that if you feel otherwise.

**Comment by mivok** [comment]
Hi David,

This appears to be because kramdown parses a fenced code block such as:

``````
```
$ foo
```
``````

as a codeblock with the contents `"$ foo\n"`, meaning there is a final, blank, line that causes markdownlint to think that there is output after the command when in fact there isn't.

The simplest fix for this is for me to ignore any blank lines (matching `/^$/` as consisting of non-command output), or I can simply strip off any trailing blank line.

Which leads me to a question for you: what do you think of the following code block, should it trigger MD014?

``````
```
$ some command

$ some command

$ some command
```
``````

**Comment by mivok** [comment]
Ignore that last question, I just realized I split lines on `\n+` which means blank lines (except for the last one) are already ignored when considering MD014, and in fact is in the current tests already.

**Comment by mivok** [comment]
So now I've actually played with this rather than just looking at the code, I'm seeing that the following does trigger MD014 in ruby markdownlint:

``````
```fence
$ foo
```
``````

but the following doesn't trigger MD014:

``````
```fence {MD014}
$ foo
```
``````

This is because if you have spaces in the first line right after the ```, kramdown will parse it as a codeblock with`fence {MD014}` as the first line rather than as the language specification.

I'm tempted to just remove the `{MD014}` from the same line of the tests and use `{MD014:linenumber}` at the bottom of the file.

Interestingly, github appears to not parse a fenced code block with spaces on the first line as a code block at all:

```ruby {MD014}
test

```
```

**Comment by DavidAnson** [comment]
That makes sense, thank you for the details! I'll go ahead and switch over the Node implementation to use the same test file you ended up with, just so they stay in sync.

**Comment by MindFireGPSDevEleSearch** [comment]
8601a4d7283eaa23b10b59cd08fe5820d80178cc

**Comment by MindFireGPSDevEleSearch** [comment]
41fc308f0d7f2647f0ae2c2d3826f48e42d964f4

**Comment by jaymzh** [comment]
This is fixed on the main branch
